### PR TITLE
Snehal-Workshop step 2 and 3

### PR DESCRIPTION
**Pull request by SnehalNaik-DEV**

### CODE REVIEW

File: ui/actions/nuxeo-favorites-toggle-button.js [hunk 1]
```diff
@@ -147,25 +147,29 @@ import '../nuxeo-button-styles.js';
...
         this.$.opAdd.execute().then(() => {
-          this.dispatchEvent(
-            new CustomEvent('added-to-favorites', {
-              composed: true,
-              bubbles: true,
-              detail: { doc: this.document },
-            }),
-          );
-          this._setFavorite(true);
+          if (window.confirm('Are you sure you want to add this document to Favorites?') === true) {
```
Comment (Gabez0r) [inline ui/actions/nuxeo-favorites-toggle-button.js]: This label should be internationalized. It mustn't be hardcoded.

File: ui/actions/nuxeo-favorites-toggle-button.js [hunk 2]
```diff
@@ -147,25 +147,29 @@ import '../nuxeo-button-styles.js';
...
         this.$.opRemove.execute().then(() => {
-          this.dispatchEvent(
-            new CustomEvent('removed-from-favorites', {
-              composed: true,
-              bubbles: true,
-              detail: { doc: this.document },
-            }),
-          );
-          this._setFavorite(false);
+          if (window.confirm('Are you sure you want to remove this document from Favorites?') === true) {
```
Comment (Gabez0r) [inline ui/actions/nuxeo-favorites-toggle-button.js]: Same here. This label should be internationalized.

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 3]
```diff
@@ -48,13 +52,22 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+        window.confirm.restore();
+      }
+    });
+
     test('it should display the document as favorite', () => {
       expect(element.favorite).to.be.true;
     });
 
-    test('toggle should remove document from favorites', async () => {
+    test('toggle should remove document from favorites when clicked ok', async () => {
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:64]: We should also make sure that if the user clicks cancel nothing happens.

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 4]
```diff
@@ -84,13 +97,20 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+        window.confirm.restore();
+      }
+    });
 
     test('it should display the document as not favorite', () => {
       expect(element.favorite).to.be.false;
     });
 
-    test('toggle should add the document to favorites', async () => {
+    test('toggle should add the document to favorites when clicked on Ok', async () => {
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:116]: We should also test what happens if the user clicks cancel.

File: ui/actions/nuxeo-favorites-toggle-button.js (hunk 1 above)
Comment (richardsd) [reply ui/actions/nuxeo-favorites-toggle-button.js]: Please check the wiki page if you need more details about it.

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 5]
```diff
@@ -48,12 +49,19 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+        window.confirm.restore();
+      }
+    });
+
     test('it should display the document as favorite', () => {
       expect(element.favorite).to.be.true;
     });
 
-    test('toggle should remove document from favorites', async () => {
+    test('toggle should remove document from favorites when clicked ok', async () => {
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:64]: We still need to test that it should not remove the document from favorites if the user clicks cancel.

**Review by Gabez0r — COMMENTED** [review]
Looking good. But we still have too many redundant commits, several with the same message. These commits should be squashed as they don't add anything meaningful to the history.

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 6]
```diff
@@ -48,16 +52,27 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+      sinon.stub(window, 'confirm').returns(true);
       tap(element);
       await waitChanged(element, 'favorite');
       expect(element.favorite).to.be.false;
     });
+
+    test('toggle should not remove document from favorites when clicked Cancel', async () => {
+      sinon.stub(window, 'confirm').returns(false);
+      expect(element.favorite).to.be.true;
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:78]: The button was not clicked, so it's normal that this wouldn't change.

File: ui/test/nuxeo-favorites-toggle-button.test.js [hunk 7]
```diff
@@ -84,16 +99,26 @@ suite('nuxeo-favorites-toggle-button', () => {
...
+    test('toggle should add the document to favorites when clicked on Ok', async () => {
+      sinon.stub(window, 'confirm').returns(true);
       tap(element);
       await waitChanged(element, 'favorite');
       expect(element.favorite).to.be.true;
     });
+
+    test('toggle should not add the document to favorites when clicked on Cancel', async () => {
+      sinon.stub(window, 'confirm').returns(false);
+      expect(element.favorite).to.be.false;
```
Comment (Gabez0r) [inline ui/test/nuxeo-favorites-toggle-button.test.js:130]: The button was not clicked, so it's normal that this wouldn't change.

**Review by Gabez0r — COMMENTED** [review]
Please see the latest comments in https://github.com/nuxeo/nuxeo-elements/pull/539

**Review by Gabez0r — COMMENTED** [review]
Good job.

File: ui/i18n/messages.json [hunk 8]
```diff
@@ -278,6 +280,7 @@
   "shareButton.operation.copied": "Link Copied",
   "shareButton.operation.copy": "Copy Link",
   "shareButton.tooltip": "Share",
+  "parentInspectorButton.tooltip": "Parent Inspector",
```
Comment (Gabez0r) [inline ui/i18n/messages.json]: Where is this label being used? It it is not used in nuxeo elements but only on nuxeo web ui, it should reside in the nuxeo web ui repository.

**Review by Gabez0r — COMMENTED** [review]
Good work!

File: ui/i18n/messages-cs.json [hunk 9]
```diff
@@ -278,6 +278,7 @@
   "shareButton.operation.copied": "Link Copied",
   "shareButton.operation.copy": "Copy Link",
   "shareButton.tooltip": "Share",
+  "parentInspectorButton.tooltip": "Parent Inspector",
```
Comment (richardsd) [inline ui/i18n/messages-cs.json]: Where is this coming from? Did you add this manually? Notice that you should only add the label to `ui/i18n/messages.json` as the translation process is done on Crowdin (check the wiki for details about the process).

**Review by richardsd — COMMENTED** [review]
Looking good 🙂 Please review https://github.com/nuxeo/nuxeo-elements/runs/7106930065?check_suite_focus=true as the PR check is failing at the moment due to linting issues.

**Pull request closed without merging**